### PR TITLE
Use Result in the network layer.

### PR DESCRIPTION
**Pull request by renatorodrigues**
Closes: #123

### CODE REVIEW

File: Sources/Stores/NetworkPersistableStore.swift [hunk 1]
```diff
@@ -134,48 +144,55 @@ where Network.Remote == Data, Persistence.Remote == Data  {
...
 
-            } catch let Alicerce.Network.Error.url(error as NSError) where error.domain == NSURLErrorDomain
-                && error.code == NSURLErrorCancelled {
-                    completion(nil, Error.cancelled)
-            } catch let error as Alicerce.Network.Error {
-                completion(nil, Error.network(error))
-            } catch {
-                completion(nil, Error.other(error))
+            case let .failure(error):
+
```
Comment (p4checo) [inline Sources/Stores/NetworkPersistableStore.swift]: extra newline? 💄

File: Sources/Stores/NetworkPersistableStore.swift [hunk 2]
```diff
@@ -134,48 +144,55 @@ where Network.Remote == Data, Persistence.Remote == Data  {
...
         } catch {
-            completion(nil, Error.other(error))
+            completion(.failure(.other(error)))
         }
     }
 
     // MARK: Network Methods
 
-    private func getNetworkData<R>(_ resource: R, completion: @escaping (Data?, Error?) -> ())
+    private func getNetworkData<R>(_ resource: R, completion: @escaping (Data?, Error?) -> Void)
```
Comment (p4checo) [inline Sources/Stores/NetworkPersistableStore.swift]: `completion: @escaping (Result<Data, Error>) -> Void`?

Comment (renatorodrigues) [reply Sources/Stores/NetworkPersistableStore.swift]: Well spotted! Fixing it.

Comment (renatorodrigues) [reply Sources/Stores/NetworkPersistableStore.swift]: Done. ✨

**Pull request merged**